### ISSUE TITLE
[Bug Fix/New Feature] Don't fail silently for large schemas

### ISSUE THREAD

**Issue by supertopher** [airbytehq/airbyte#12514]
## Problem Statement
If you send anything (but mostly a schema) via temporal, this request is sent under the hood using GRPC.  
GRPC by default maxed out at 4MB

## Expected work to be done
Sending a schema via setting up a new connection with a schema in excess of 4MB does not fail silently

## Potential Approaches
Increase the GRPC message length for the specific channel temporal is using
[example here](https://stackoverflow.com/questions/42629047/how-to-increase-message-size-in-grpc-using-python)

Decrease the size of the message using temporal's `system.transactionSizeLimit` (I suspect this will make the problem worse not better as it seems like this is actually a further limitation not an expansion of message size)

## You will know you are done when:
- [ ] You can create a connection with a schema over 4MB
- [ ] You see an error message in the UI that communicates a schema is too large (or make a new issue to track)

first discovered in https://github.com/airbytehq/oncall/issues/212

**Comment by cgardens** [comment]
Thanks, @supertopher! Super helpful breakdown. I'd be curious to know if you know how risky it is to do just bump the size in the shorter term in order to unblock people.

For the "right" fix what you described make sense. We could potentially store the `ConfiguredCatalog` in the db in the same way we are doing for discovered catalogs. As long as we have a way of knowing exactly what catalog we ran on, I think we are good. If it's stored elsewhere that's fine as long as it is immutable.

**Comment by pmossman** [comment]
Short term plan: Scope out how much effort it would take to throw an early error if the schema size is too big and will fail

**Comment by grishick** [comment]
Is this a duplicate of this issue: https://github.com/airbytehq/airbyte/issues/4564

**Comment by supertopher** [comment]
@grishick can we close yours?  I think we are pretty close to actually starting this work.  We have already done some work to at least fail quickly
